### PR TITLE
x509-cert: add Signed Certificate Timestamp (SCT) extension support

### PR DESCRIPTION
**Pull request by imor**
This PR adds support for [Signed Certificate Timestamp extension](https://www.rfc-editor.org/rfc/rfc6962#section-3.2) to the `x509-cert` crate. Since the structures in SCT extension are TLS encoded we need to add a dependency on the `tls_codec` crate. This dependency has been made an optional dependency which is only enabled when the newly added `sct` feature is enabled.

### CODE REVIEW

**Comment by imor** [comment]
@tnytown the only thing pending is [this TODO](https://github.com/RustCrypto/formats/blob/993ca90887f4f64db107bc05b3ed8656a86c95f4/x509-cert/Cargo.toml#L26-L29). Basically this PR is waiting for a new release of `tls_codec` which includes changes merged in #1132 and #1133. Once that is done, the path based dependency on `tls_codec` can be replaced with the new published version and then this PR will be ready to review.

@franziskuskiefer, @tarcieri when is the next planned release for `tls_codec`?

**Comment by tarcieri** [comment]
I can potentially cut a release of `tls_codec` if need be but I'd prefer for @franziskuskiefer to do it

**Comment by franziskuskiefer** [comment]
I plan to do one next week or so with couple changes. But if there's the need for an earlier one, I can do one before.

**Comment by franziskuskiefer** [comment]
> @franziskuskiefer gentle ping on the `tls_codec` release :)

I prepared the release. There are a couple small things that need to get fixed but I think I can do the release today or tomorrow latest.

File: x509-cert/src/ext/pkix/sct.rs
Comment (franziskuskiefer) [inline x509-cert/src/ext/pkix/sct.rs:76]: We renamed this version to `tls_deserialize_bytes` to disambiguate the two versions.

**Review by franziskuskiefer — COMMENTED** [review]
I release tls_codec 0.4.0. There's one breaking change there. I commented on that here. With the rename here, this PR works as is.

**Review by woodruffw — APPROVED** [review]
Not a maintainer, but I did a review based on what downstreams like `sigstore-rs` need and this LGTM!

File: x509-cert/src/ext/pkix/sct.rs
Comment (baloo) [inline x509-cert/src/ext/pkix/sct.rs:25]: TODO(baloo): make a backport of #1094 and publish a release on 0.9

File: x509-cert/src/ext/pkix/sct.rs
Comment (baloo) [inline x509-cert/src/ext/pkix/sct.rs:58]:
```suggestion
    /// Creates a new [`SignedCertificateTimestampList`] from a slice of [`SerializedSct`]s.
```

File: x509-cert/src/ext/pkix/sct.rs
Comment (baloo) [reply x509-cert/src/ext/pkix/sct.rs:223]: I believe this is https://www.rfc-editor.org/rfc/rfc8446.html#page-42

Comment (baloo) [reply x509-cert/src/ext/pkix/sct.rs:223]: I think this is mostly that: `SignatureAndHashAlgorithm` is `SignatureScheme` in TLS1.3. The first byte encodes for the hash alg, the second byte encodes for signature alg?

File: x509-cert/src/ext/pkix/sct.rs
Comment (baloo) [reply x509-cert/src/ext/pkix/sct.rs:25]: https://github.com/RustCrypto/formats/pull/1282

Comment (baloo) [reply x509-cert/src/ext/pkix/sct.rs:25]: https://docs.rs/const-oid/0.9.6/const_oid/db/rfc6962/constant.CT_PRECERT_SCTS.html

**Review by baloo — APPROVED** [review]

File: x509-cert/Cargo.toml
Comment (tarcieri) [inline x509-cert/Cargo.toml:28]: We usually do these as one-liners...

```suggestion
tls_codec = { version = "0.4.0", default-features = false, features = ["derive"], optional = true }
```

File: x509-cert/src/ext/pkix/sct.rs
Comment (tarcieri) [inline x509-cert/src/ext/pkix/sct.rs:5]: Following `std` we usually do comments first, then attributes...
```suggestion
//! Signed Certificate Timestamp list extension as defined in the
//! [Certificate Transparency RFC 6962].
//!
//! [Certificate Transparency RFC 6962]: https://datatracker.ietf.org/doc/html/rfc6962

#![cfg(feature = "sct")]
```

File: x509-cert/src/ext/pkix/sct.rs
Comment (tarcieri) [reply x509-cert/src/ext/pkix/sct.rs:26]: Will leave this in for now but we can address it as a followup

**Review by tarcieri — APPROVED** [review]

**Pull request merged**
